tools/src/main/java/com/nedap/archie/archetypevalidator/ArchetypeValidator.java

```diff
@@ -142,10 +140,9 @@ public ValidationResult validate(Archetype archetype, FullArchetypeRepository re
         //we assume we always want a new validation to be run, for example because the archetype
         //has been updated. Therefore, do not retrieve the old result from the repository
         archetype = cloneAndPreprocess(combinedModels, archetype);//this clones the actual archetype so the source does not get changed
-        ValidationResult parentValidationResult = null;
         Archetype flatParent = null;
         if(archetype.isSpecialized()) {
-            parentValidationResult = getParentValidationResult(archetype, repository);
+            ValidationResult parentValidationResult = getValidationResult(archetype.getParentArchetypeId(), repository);
```

Comment:
This means calling `ArchetypeRepository::getArchetype` with a null archetype-id. The archetype repository is an interface, and its documentation does not specify the behaviour of calling it with a null-parameter. This means it is possible that implementations exist that will fail after this change.
Perhaps change `getValidationResult()` to return null in case archetype ID is null?

Reply:
Oh, wait, that was already the case before. Still, could be good to add the check - this is a validator, and a valid specialised archetype must have an archetype id - but the input is not guaranteed to be valid, so best not to potentially throw an exception?

Reply:
`archetype.isSpecialized()` is true if and only if `archetype.getParentArchetypeId()` is non-null, so that can't happen.

Reply:
You're right. Ok, approving